festim/hydrogen_transport_problem.py

```diff
@@ -158,6 +160,37 @@ def define_markers_and_measures(self):
             "dx", domain=self.mesh.mesh, subdomain_data=self.volume_meshtags
         )
 
+    def check_borders(self, mesh):
+        """Checks that the borders of the subdomain are within the domain
+
+        Args:
+            mesh (festim.Mesh): the mesh of the model
+
+        Raises:
+            Value error: if borders outside the domain
+        """
+        # check that subdomains are connected
+        all_borders = []
+        for vol in self.volume_subdomains:
+            for border in vol.borders:
+                all_borders.append(border)
```

Comment:
We can make this a one liner if we want to
```suggestion
        all_borders = [border for border in vol.borders for vol in self.volume_subdomains]
```